### PR TITLE
Integrate sds with launcher

### PR DESCRIPTION
**Pull request by kylelawsonAND**
### What is the context of this PR?
As part of the Prepop feature, it is necessary for SDS Dataset Metadata to be selected within launcher from a set of available SDS dataset metadatas for a given survey. 
This PR support the changes to launcher outlined in [this PR ](https://github.com/ONSdigital/eq-questionnaire-launcher/pull/49). 

### Changes
Mock SDS: 
The mock SDS has been updated to expose an additional endpoint `/v1/dataset_metadata` which will be called by launcher to acquire a list of available SDS dataset metadatas to be selected by the dev/tester. This endpoint expects survey id and period id as url params, and loads mock datasets for survey ids 123 and 456 which correspond to the schemas outlined below. 

Schemas: 
To support testing for multiple sets of datasets per survey a new schema `test_supplementary_data_multiple_datasets` has been added which corresponds to a response with multiple available datasets when its survey id is provided to the new mock endpoint. The original schema `test_supplementary_data` has been modified to include the survey id as a field in metadata, as this is what is expected from RM and how runner extracts the survey id for making the SDS request. 

Mock Data: 
Another supplementary data payload has been added `supplementary_data_with_repeat_v2.json` which features a repeat, this exists so its ID along with the ID of the other supplementary data payload with a repeat `supplementary_data_with_repeat.json` can be sent as a response from the new dataset endpoint to demonstrate multiple datasets being available, not just one. 
`supplementary_dataset_single_item_response.json` is the response payload from the new mock endpoint when provided the survey id from `test_supplementary_data`. This details only one supplementary data option `supplementary_data_no_repeat.json`.
Likewise `supplementary_dataset_multiple_item_response.json` is the response for the response when the survey id from `test_supplementary_data_multiple_datasets` is provided. This details the supplementary data options `supplementary_data_with_repeat` and `supplementary_data_with_repeat_v2`. 

### How to test functionally

1. Clone and run the [launcher PR](https://github.com/ONSdigital/eq-questionnaire-launcher/pull/49) with `go run launch.go`
2. Run the supporting services for runner with `make dev-compose-up` as normal (delete the launcher container if that is built despite launcher already being run) 
3. Launch the mock SDS from a terminal in runner`python -m scripts.mock_sds_endpoint`
4. Launch runner as normal.
5. Open launcher in chrome
6. Select V2 Launch Pattern and the schema `test_supplementary_data`
7. Survey Metadata should load as normal, with survey id 123 and a new sds_dataset_id dropdown with only one option. 
8. The first and only SDS dataset metadata should have loaded and displayed below.
9. The survey should be launchable.
10. Open launcher again
11. Select V2 Launch Pattern and the schema `test_supplementary_data_with_multiple_datasets`
12. Survey Metadata should load as normal, with survey id 456 and 2 `sds_dataset_ids` in the `sds_dataset_id` dropdown. The first one will auto-select. 
13. Swap between the given options and the Dataset Metadata section should update to reflect the metadata. 
14. Launch the survey with both options to verify it launches correctly. 
15. Open launcher again
16. Select V2 Launch Pattern and either `test_supplementary_data` or `test_supplementary_data_with_multiple_datasets`.
17. Enter a random number for the survey id in the survey metadata section.
18. The Dataset Metadata section should disappear and the `sds_dataset_id` dropdown should clear of options, and the Launch Survey button should not be available, as the survey requires an `sds_dataset_id` value.  
19. In step 17, using the id of the schema you did not load, will load the available datasets for that survey but it will be an invalid dataset for the selected survey and thus crash on launch. 

### How to review
Refer to the [Launcher PR](https://github.com/ONSdigital/eq-questionnaire-launcher/pull/49)

### CODE REVIEW

File: scripts/mock_data/supplementary_dataset_single_item_response.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[
```
Comment (MebinAbraham) [inline scripts/mock_data/supplementary_dataset_single_item_response.json:1]: I think filename should include `metadata` to signify what this is used for. Same for the one below.

File: scripts/mock_data/supplementary_data_with_repeat_v2.json [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+          "guidance_include": {
+            "title": "Include",
+            "list": [
+              "pots and pans",
+              "chopping board"
+            ]
+          },
+          "guidance_exclude": {
+            "title": "Exclude",
+            "list": []
```
Comment (MebinAbraham) [inline scripts/mock_data/supplementary_data_with_repeat_v2.json:95]: We shouldn't have empty list here.

File: schemas/test/en/test_supplementary_data_multiple_datasets.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{
```
Comment (MebinAbraham) [inline schemas/test/en/test_supplementary_data_multiple_datasets.json:1]: Do we really need this schema?. Adding a runner schema just to test launcher doesn't seem worth it. Just have multiple for the main one, that proves both cases really?

Comment (katie-gardner) [reply schemas/test/en/test_supplementary_data_multiple_datasets.json:1]: Yep thats a good point - we can just change the survey_id in metadata to see the one with multiple datasets come through and have the single schema already there

File: scripts/mock_data/supplementary_dataset_metadata_response.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  },
+  {
+    "survey_id": "123",
+    "period_id": "ttt",
+    "title": "Supplementary data with repeating items",
+    "sds_schema_version": 1,
+    "sds_published_at": "2022-03-01T12:30:02Z",
+    "total_reporting_units": 2,
+    "schema_version": "v1.0.0",
+    "sds_dataset_version": 1,
```
Comment (MebinAbraham) [inline scripts/mock_data/supplementary_dataset_metadata_response.json:22]: This `sds_dataset_version` should be 2, and the next one 3 to make it realistic data.

File: scripts/mock_data/supplementary_dataset_metadata_response.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
+[
+  {
+    "survey_id": "123",
+    "period_id": "ttt",
```
[the remark this answers is not in the document]
Comment (kylelawsonAND) [reply scripts/mock_data/supplementary_dataset_metadata_response.json]: Done 👍

File: scripts/mock_data/supplementary_dataset_metadata_response.json (hunk 4 above)
Comment (kylelawsonAND) [reply scripts/mock_data/supplementary_dataset_metadata_response.json]: Done 👍

**Review by MebinAbraham — COMMENTED** [review]
The run mock sds markdown has a list of dataset IDs. I think we can remove it and add a bit around, it will be preloaded via launcher etc.

File: scripts/mock_data/supplementary_dataset_metadata_response.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  },
+  {
+    "survey_id": "123",
+    "period_id": "202302",
+    "title": "Supplementary data with repeating items",
+    "sds_schema_version": 1,
+    "sds_published_at": "2022-03-01T12:30:02Z",
+    "total_reporting_units": 2,
+    "schema_version": "v1.0.0",
+    "sds_dataset_version": 2,
```
Comment (MebinAbraham) [inline scripts/mock_data/supplementary_dataset_metadata_response.json:22]: The sds_dataset_version would only be 2 if it was the same period. So either set all to same period, or set all version to 1 given period is all diff.

**Review by berroar — APPROVED** [review]
LGTM other than updating the readme 👍

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**